addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -918,6 +918,69 @@ def test_message_route_alias_multiple_new(self):
         new_simple = self.env['mail.test.gateway'].search([('name', '=', 'Test Subject')])
         self.assertEqual(len(new_simple), 1, 'message_process: a new mail.test should have been created')
 
+    def test_message_route_alias_reply_author_notify(self):
+        """Make sure users are notified when a reply is sent to an alias address.
+
+        The alias owner should impact who the creator of the message is but not the notifications.
+        """
+
+        test_record = self.env['mail.test.ticket'].create({})
+        author_partner = self.env['res.partner'].create({
+            'name': self.email_from,
+            'email': self.email_from,
+        })
+        message = self.env['mail.message'].create({
+            'body': '<p>test</p>',
+            'email_from': 'reply_author_notify_test@test.com',
+            'message_type': 'email',
+            'model': test_record._name,
+            'res_id': test_record.id,
+        })
+        Model = self.env['ir.model']._get(test_record._name)
+        self.env['mail.alias'].create({
+            'alias_model_id': Model.id,
+            'alias_name': 'reply_author_notify_test',
+            'alias_user_id': self.user_employee.id
+        })
+
+        test_record.message_subscribe((author_partner | self.user_employee.partner_id).ids)
+
+        messages = test_record.message_ids
+
+        odoobot = self.env.ref('base.user_root')
+        self.assertFalse(odoobot.active, 'notification logic relies on odoobot being archived')
+
+        users = [self.user_employee, odoobot]
+        email_tos = ['reply_author_notify_test@test.com', 'some_non_aliased_email@test.com']
+        params = ((email_to, user)
+                  for user in users
+                  for email_to in email_tos)
+        for email_to, user in params:
+            with self.subTest(user=user, email_to=email_to):
+                with self.mock_mail_gateway(), self.mock_mail_app():
+                    self.format_and_process(
+                        MAIL_TEMPLATE, self.email_from, email_to,
+                        subject=message.message_id, extra=f'In-Reply-To:\r\n\t{message.message_id}\n',
+                        model=None, user=user)
```

Comment:
Not sure what user is doing here ? I think it ends up in formatting of email as available rendering variable, not sure it does more ?

Reply:
Ooh right, I had a modification to format_and_process to have with_user and with_context in there and I removed when I removed the context :facepalm: 

I'll add it back, and use a different name